pyramid_openapi3/exceptions.py

```diff
@@ -35,5 +34,72 @@ def __str__(self) -> str:
         """Return str(self.detail) or self.explanation."""
         return str(self.detail) if self.detail else self.explanation
 
-    def _json_formatter(self, status, body, title, environ) -> t.Dict:
-        return {"message": body, "code": status, "title": self.title}
+
+def extract_error(err: OpenAPIError) -> t.Dict[str, str]:
+    """Extract error's JSON response.
+
+    You can tell pyramid_openapi3 to use your own version of this
+    function by providing a dotted-name to your function in
+    `request.registry.settings["pyramid_openapi3_extract_error"]`.
+
+    This function expects the below definition in openapi.yaml
+    file. If your openapi.yaml is different, you have to
+    provide your own extract_error() function.
+
+    ```
+    components:
+
+      schemas:
+
+        Error:
+          type: object
+          required:
+            - exception
+            - message
+          properties:
+            field:
+              type: string
+            message:
+              type: string
+            exception:
+              type: string
+
+      responses:
+
+        ValidationError:
+          description: OpenAPI request/response validation failed
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/Error"
+    ```
+    """
+    if getattr(err, "schema_errors", None) is not None:
+        for schema_error in err.schema_errors:  # pragma: no branch
+            return extract_error(schema_error)
```

Comment:
Should it be `return [extract_error(schema_error) for schema_error in err.schema_errors]`?

Reply:
most likely yes ... or `yield extract_error(schema_error)` instead of `return`

Reply:
No, not really, that makes it wrapped in double list in case of recursion.

Reply:
I didn't check what `err.schema_errors` looks like but the code reads like that:
The loop iterates over `err.schema_errors` and returns the first one in this list. In case there are more items in `err.schema_errors` they get ignored.

Reply:
Ok, dug around a bit. `schema_errors` seems to be defined only on [InvalidSchemaValue](https://github.com/p1c2u/openapi-core/blob/cb54400136b0e335780dc7eaff769ed3be879115/openapi_core/unmarshalling/schemas/exceptions.py#L30) exceptions.  
I think in this case it really would be interesting to see the full list of errors that caused an `InvalidSchemaValue` error.

Reply:
Right. I also think it should return a list instead of dict. I believe other schema parsers also use list. e.g. https://github.com/marshmallow-code/marshmallow/blob/dev/src/marshmallow/exceptions.py#L39